distribution/push_v2.go

```
@@ -186,6 +187,15 @@ func (p *v2Pusher) pushV2Tag(ctx context.Context, ref reference.NamedTagged, id
 			return err
 		}
 
+		// This is a temporary environment variables used in CI to allow pushing
+		// manifest v2 schema 1 images to test-registries used for testing *pulling*
+		// these images.
+		if os.Getenv("DOCKER_ALLOW_SCHEMA1_PUSH") == "" {
+			msg := "[DEPRECATED] support for manifest v2 schema1 has been removed. More information at https://docs.docker.com/registry/spec/deprecated-schema-v1/"
+			logrus.Error(msg)
+			return errors.New(msg)
```

We should return the original error. Can wrap it with removal message.